src/components/Layout/Header/NavBar/KeepKey/ChangeLabel.tsx

```diff
@@ -0,0 +1,67 @@
+import { Button, Flex, Input, useColorModeValue } from '@chakra-ui/react'
+import { useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { AwaitKeepKey } from 'components/Layout/Header/NavBar/KeepKey/AwaitKeepKey'
+import { LastDeviceInteractionStatus } from 'components/Layout/Header/NavBar/KeepKey/LastDeviceInteractionStatus'
+import { SubmenuHeader } from 'components/Layout/Header/NavBar/SubmenuHeader'
+import { useKeepKey } from 'context/WalletProvider/KeepKeyProvider'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { SubMenuBody } from '../SubMenuBody'
+import { SubMenuContainer } from '../SubMenuContainer'
+
+export const ChangeLabel = () => {
+  const translate = useTranslate()
+  const { state } = useWallet()
+  const { walletInfo } = state
+  const { keepKeyWallet } = useKeepKey()
+  const {
+    state: { awaitingDeviceInteraction }
+  } = useWallet()
+  const [keepKeyLabel, setKeepKeyLabel] = useState(walletInfo?.name)
+
+  const handleChangeLabelInitializeEvent = async () => {
+    await keepKeyWallet?.applySettings({ label: keepKeyLabel })
```

Comment:
possible `UnhandledPromiseRejection` error

Reply:
Handled in https://github.com/shapeshift/web/pull/1341/commits/f9fe9b56dee0d189360272b2cc4ae6e9a2d20f34